test/boost/sstable_test.hh

```diff
@@ -32,18 +32,16 @@ class column_family_test {
 public:
     column_family_test(lw_shared_ptr<replica::column_family> cf) : _cf(cf) {}
 
+    // NOTE: must run in a thread.
     void add_sstable(sstables::shared_sstable sstable) {
-        _cf->_main_sstables->insert(std::move(sstable));
-        _cf->refresh_compound_sstable_set();
+        auto new_sstables = { sstable };
+        _cf->as_table_state().on_compaction_completion(sstables::compaction_completion_desc{ .new_sstables = new_sstables }, sstables::offstrategy::no).get();
     }
 
     // NOTE: must run in a thread
     void rebuild_sstable_list(const std::vector<sstables::shared_sstable>& new_sstables,
             const std::vector<sstables::shared_sstable>& sstables_to_remove) {
-        auto permit = seastar::get_units(_cf->_sstable_set_mutation_sem, 1).get0();
-        auto builder = replica::table::sstable_list_builder(std::move(permit));
-        _cf->_main_sstables = builder.build_new_list(*_cf->_main_sstables, _cf->_compaction_strategy.make_sstable_set(_cf->schema()), new_sstables, sstables_to_remove).get0();
-        _cf->refresh_compound_sstable_set();
+        _cf->as_table_state().on_compaction_completion(sstables::compaction_completion_desc{ .old_sstables = sstables_to_remove, .new_sstables = new_sstables }, sstables::offstrategy::no).get();
```

Comment:
This change seems intrusive enough to be separated into its own patch too.

Reply:
All the changes in "test: compaction: Prepare tests for compaction group changes" were done as a consequence of these change in column_family_test. That's because those functions now have to run in a thread. But I can prepare the tests for the preparation of the tests :-)